html/modules/custom/reliefweb_analytics/src/Command/ReliefwebMostReadCommand.php

```diff
@@ -0,0 +1,468 @@
+    $parameters = $this->getHomePagePayload();
+    $data = $this->fetchGa4Data($parameters);
+    if (!empty($data)) {
+      $results['front'] = [
+        'front',
+        implode(',', $data),
+      ];
+    }
+
+    if (!empty($results)) {
+      $this->updateCsv($results);
+    }
+  }
+
+  /**
+   * Most read for countries.
+   *
+   * @command reliefweb_analytics:countries
+   * @usage reliefweb_analytics:countries
+   *   Send emails.
+   * @validate-module-enabled reliefweb_analytics
+   * @aliases reliefweb-mostread-countries
+   */
+  public function countries() {
+    $results = [];
+
+    // Load all terms.
+    $query = $this->entityTypeManager->getStorage('taxonomy_term')->getQuery();
+    $query->condition('vid', 'country');
+    $tids = $query->execute();
+    $countries = $this->entityTypeManager->getStorage('taxonomy_term')->loadMultiple($tids);
+
+    // Fetch data from GA4.
+    foreach ($countries as $country) {
+      $this->logger()->notice('Processing ' . $country->label());
+      $parameters = $this->getCountryPayload($country->label());
+      $data = $this->fetchGa4Data($parameters);
+      if (!empty($data)) {
+        $results[$country->id()] = [
+          $country->id(),
+          implode(',', $data),
+        ];
+      }
+    }
+
+    if (!empty($results)) {
+      $this->updateCsv($results);
+    }
+  }
+
+  /**
+   * Most read for disasters.
+   *
+   * @command reliefweb_analytics:disasters
+   * @usage reliefweb_analytics:disasters
+   *   Send emails.
+   * @validate-module-enabled reliefweb_analytics
+   * @aliases reliefweb-mostread-disasters
+   */
+  public function disasters() {
+    // Load terms having a job URL.
+    $query = $this->entityTypeManager->getStorage('taxonomy_term')->getQuery();
+    $query->condition('vid', 'disaster');
+    $tids = $query->execute();
+    $disasters = $this->entityTypeManager->getStorage('taxonomy_term')->loadMultiple($tids);
+    foreach ($disasters as $disaster) {
+      $this->logger()->notice('Processing ' . $disaster->label());
+      $parameters = $this->getDisasterPayload($disaster->label());
+      $data = $this->fetchGa4Data($parameters);
+      if (!empty($data)) {
+        $results[$disaster->id()] = [
+          $disaster->id(),
+          implode(',', $data),
+        ];
+      }
+    }
+
+    if (!empty($results)) {
+      $this->updateCsv($results);
+    }
+  }
+
+  /**
+   * Most read for disasters.
+   *
+   * @command reliefweb_analytics:disasters-all
+   * @usage reliefweb_analytics:disasters-all
+   *   Send emails.
+   * @validate-module-enabled reliefweb_analytics
+   * @aliases reliefweb-mostread-disasters-all
+   */
+  public function disastersAll() {
+    $results = [];
+    $parameters = $this->getAllDisastersPayload();
+    $combined = $this->fetchGa4DataCombined($parameters);
+
+    if (!empty($combined)) {
+      foreach ($combined as $disaster => $data) {
+        $results[$disaster] = [
```

Comment:
Isn't `disaster` here the disaster label as `getAllDisastersPayload` uses the `content_report_disaster` dimension? If that's the case, I'm guessing there should be some lookup to do to retrieve the ID?